pyproject.toml

```diff
@@ -141,6 +142,22 @@ dev = [
     "vulture>=1.0",
 ]
 
+[tool.flake8]
+max-complexity = 91
```

Comment:
@cclauss just curious why you added this max-complexity?
This one was not in original file.

Reply:
This function is 9 times more complex than the suggested code complexity setting.  This is also what DeepSource is complaining about but they have a different calculation than flake8 and ruff.

It is useful to set upper limits on complexity so that if someone makes this code even more complex (i.e. less understandable and more difficult to maintain) then the changing of this setting should set off a discussion in the code review about taking time to refactor or even add more tests.

Reply:
Yes, not the best software engineering example of logic. But I don't see value adding a super high complexity to it.

I would prefer to set a normal recommended value and ignore this method with a TODO to reduce the complexity and remove the ignore.

Reply:
```suggestion
```
Let’s remove it here and I will do as you say in a separate code complexity PR using ruff after that lands.